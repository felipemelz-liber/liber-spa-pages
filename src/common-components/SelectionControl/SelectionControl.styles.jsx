import styled from 'styled-components';
import { handleThemeFromObject as getTheme } from 'liber-components';
import RawIcon from '@mdi/react';

export const Container = styled.div`
  display: flex;
  flex-direction: row;
  padding: 16px;
  align-items: center;
`;

export const SelectedText = styled.span`
  font-family: Roboto;
  font-size: 16px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.38;
  letter-spacing: 0.06px;
  text-align: left;
  color: #405f71;
  display: flex;
  align-items: center;
`;

export const Icon = styled(RawIcon)`
  margin-right: 8px;
  width: 16px;
  height: 16px;
`;

export const SelectActions = styled(SelectedText)`
  color: ${({ theme }) => getTheme(theme, 'colors.primary', '#009dff')};
  margin-left: 24px;
  cursor: pointer;
  & > ${Icon} {
    fill: ${({ theme }) => getTheme(theme, 'colors.primary', '#009dff')};
  }
`;

export const Spacer = styled.div`
  flex-grow: 1;
`;
