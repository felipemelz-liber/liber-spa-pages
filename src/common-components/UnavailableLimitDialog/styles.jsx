import styled from 'styled-components';
import { FlatButton } from 'liber-components';

const TEXT_COLOR = '#6287a7';

export const Container = styled.div`
  padding: 24px 24px 12px 24px;
  width: 480px;

  * {
    font-family: Roboto;
  }
`;

export const Header = styled.div`
  font-size: 21px;
  font-weight: bold;
  line-height: 1.4;
  letter-spacing: 0.08px;
  color: ${TEXT_COLOR};
  margin-bottom: 24px;
`;

export const Content = styled.div`
  font-size: 16px;
  line-height: 1.38;
  letter-spacing: 0.01px;
  margin-bottom: 24px;
  color: ${TEXT_COLOR};
  margin-bottom: 24px;
`;

export const ActionsRow = styled.div`
  display: flex;
  justify-content: flex-end;
`;

export const Button = styled(FlatButton).attrs(() => ({ size: 'large' }))``;
