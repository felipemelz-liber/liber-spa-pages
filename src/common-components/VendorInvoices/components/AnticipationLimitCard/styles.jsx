import styled from 'styled-components';

export const Card = styled.div`
  font-family: Roboto;
  color: #6287a7;
  width: 360px;
  box-shadow: 0px 3px 6px #00000029;
  border: 1px solid #cdd7df;
  border-radius: 8px;
  padding: 16px 24px;
  margin-bottom: 56px;

  * {
    font-family: Roboto;
  }
`;

export const CardTitle = styled.div`
  font-size: 16px;
  line-height: 22px;
  letter-spacing: 0.06px;
  font-weight: 500;
  color: #4b6f85;
  margin-bottom: 4px;
`;
