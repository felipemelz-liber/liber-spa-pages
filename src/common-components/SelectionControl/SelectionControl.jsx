import React from 'react';
import PropTypes from 'prop-types';
import { mdiBroom, mdiCheckAll } from '@mdi/js';
import { Container, SelectedText, SelectActions, Icon, Spacer } from './SelectionControl.styles';
import { dispatchDataLayerEvent } from '../../vendor/Utils';

const getSelectedText = selectionProps => {
  const { selectedItems, unselectedItems, hasSelectedAllItems, itemsCount } = selectionProps;
  const selectedItemsCount = selectedItems.length;

  if (hasSelectedAllItems) {
    return unselectedItems.length === 0 ? `Todos os ${itemsCount}` : selectedItemsCount;
  }

  return selectedItemsCount;
};

export const SelectionControl = props => {
  const {
    unselectedItems,
    hasSelectedAllItems,
    itemsCount,
    selectedItems,
    selectAll,
    clearSelection,
    itemName,
    actions,
    isAdmin,
  } = props;

  const hasAllInvoicesSelected =
    (hasSelectedAllItems && unselectedItems.length === 0) || selectedItems.length === itemsCount;

  const { singular, plural } = itemName;

  const isPlural = () => hasSelectedAllItems || selectedItems.length > 1;

  const handleSelectAllClick = () => {
    dispatchDataLayerEvent('titulosSelecionarTodos', isAdmin);

    selectAll();
  };

  const handleClearSelectionClick = () => {
    dispatchDataLayerEvent('titulosLimparSelecao', isAdmin);

    clearSelection();
  };

  return hasSelectedAllItems || selectedItems.length > 0 ? (
    <Container>
      <SelectedText>{`${getSelectedText(props)} ${
        isPlural() ? `${plural} foram selecionados` : `${singular} foi selecionado`
      }`}</SelectedText>
      {!hasAllInvoicesSelected ? (
        <SelectActions onClick={handleSelectAllClick}>
          <Icon path={mdiCheckAll} />
          {`Selecionar todos os ${itemsCount} ${plural}`}
        </SelectActions>
      ) : null}
      <SelectActions onClick={handleClearSelectionClick}>
        <Icon path={mdiBroom} /> Limpar seleção
      </SelectActions>
      <Spacer />
      {actions}
    </Container>
  ) : null;
};

SelectionControl.propTypes = {
  selectedItems: PropTypes.arrayOf(PropTypes.number),
  unselectedItems: PropTypes.arrayOf(PropTypes.object),
  hasSelectedAllItems: PropTypes.bool,
  itemsCount: PropTypes.number,
  selectAll: PropTypes.func,
  clearSelection: PropTypes.func,
  itemName: PropTypes.shape({
    singular: PropTypes.string,
    plural: PropTypes.string,
  }),
  actions: PropTypes.node,
  isAdmin: PropTypes.bool,
};

SelectionControl.defaultProps = {
  selectedItems: [],
  unselectedItems: [],
  hasSelectedAllItems: false,
  itemsCount: 0,
  selectAll: () => {},
  clearSelection: () => {},
  itemName: {
    singular: 'item',
    plural: 'itens',
  },
  actions: null,
  isAdmin: false,
};

export default SelectionControl;
