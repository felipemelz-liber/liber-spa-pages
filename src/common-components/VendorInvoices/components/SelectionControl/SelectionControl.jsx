import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { quoteAllInvoices, resetInvoices } from '../../actions/VendorInvoicesActions';
import { SelectionControl as BaseControl } from '../../../SelectionControl/SelectionControl';

export const SelectionControl = props => {
  const {
    hasSelectedAllItems,
    unselectedInvoices: unselectedItems,
    invoicesCount: itemsCount,
    selectedInvoices: selectedItems,
    selectAllInvoices: selectAll,
    clearInvoices: clearSelection,
  } = props;

  const mappedProps = {
    hasSelectedAllItems,
    unselectedItems,
    itemsCount,
    selectedItems,
    selectAll,
    clearSelection,
  };

  const itemName = {
    singular: 'título',
    plural: 'títulos',
  };

  return <BaseControl {...mappedProps} itemName={itemName} />;
};

SelectionControl.propTypes = {
  selectedInvoices: PropTypes.arrayOf(PropTypes.number),
  unselectedInvoices: PropTypes.arrayOf(PropTypes.object),
  hasSelectedAllItems: PropTypes.bool,
  invoicesCount: PropTypes.number,
  selectAllInvoices: PropTypes.func,
  clearInvoices: PropTypes.func,
};
SelectionControl.defaultProps = {
  selectedInvoices: [],
  unselectedInvoices: [],
  hasSelectedAllItems: false,
  invoicesCount: 0,
  selectAllInvoices: () => {},
  clearInvoices: () => {},
};

const mapStateToProps = ({
  vendorInvoices: {
    pagination: { count: invoicesCount },
    hasSelectedAllItems,
    selectedInvoices,
    unselectedInvoices,
  },
}) => ({ invoicesCount, selectedInvoices, hasSelectedAllItems, unselectedInvoices });

const mapDispatchToProps = {
  selectAllInvoices: quoteAllInvoices,
  clearInvoices: resetInvoices,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(SelectionControl);
