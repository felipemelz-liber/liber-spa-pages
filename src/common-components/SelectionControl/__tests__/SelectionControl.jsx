import React from 'react';
import sinon from 'sinon';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { SelectionControl } from '../SelectionControl';
import { SelectActions } from '../SelectionControl.styles';

describe('SelectionControl tests', () => {
  configure({ adapter: new Adapter() });

  const selectAll = sinon.spy();
  const clearSelection = sinon.spy();

  const spyProps = {
    selectAll,
    clearSelection,
  };

  const fewSelectedProps = {
    selectedItems: [1, 2, 3],
    itemsCount: 10,
    ...spyProps,
  };

  const almostAllSelectedProps = {
    selectedItems: [1, 2, 3],
    itemsCount: 4,
    unselectedItems: [{ id: 4 }],
    hasSelectedAllItems: true,
    ...spyProps,
  };

  const allSelectedProps = {
    selectedItems: [1, 2, 3],
    itemsCount: 3,
    unselectedItems: [],
    hasSelectedAllItems: true,
    ...spyProps,
  };

  it('should render nothing with no props', () => {
    const wrapper = shallow(<SelectionControl />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly with some selected items', () => {
    const wrapper = shallow(<SelectionControl {...fewSelectedProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly with almost all selected items', () => {
    const wrapper = shallow(<SelectionControl {...almostAllSelectedProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly with all selected items', () => {
    const wrapper = shallow(<SelectionControl {...allSelectedProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call clearSelection on button clear click', () => {
    const wrapper = shallow(<SelectionControl {...fewSelectedProps} />);
    const clearButton = wrapper.find(SelectActions).at(1);

    clearButton.simulate('click');

    expect(clearSelection.called).toBeTruthy();
  });

  it('should call selectAll on button select all click', () => {
    const wrapper = shallow(<SelectionControl {...fewSelectedProps} />);
    const selectButton = wrapper.find(SelectActions).at(0);

    selectButton.simulate('click');

    expect(selectAll.called).toBeTruthy();
  });
});
