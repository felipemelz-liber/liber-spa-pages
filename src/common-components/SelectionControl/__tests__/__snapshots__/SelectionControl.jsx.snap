// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectionControl tests should render correctly with all selected items 1`] = `
<styled.div>
  <styled.span>
    Todos os 3 itens foram selecionados
  </styled.span>
  <Styled(styled.span)
    onClick={[Function]}
  >
    <Styled(Icon)
      path="M19.36,2.72L20.78,4.14L15.06,9.85C16.13,11.39 16.28,13.24 15.38,14.44L9.06,8.12C10.26,7.22 12.11,7.37 13.65,8.44L19.36,2.72M5.93,17.57C3.92,15.56 2.69,13.16 2.35,10.92L7.23,8.83L14.67,16.27L12.58,21.15C10.34,20.81 7.94,19.58 5.93,17.57Z"
    />
     Limpar seleção
  </Styled(styled.span)>
  <styled.div />
</styled.div>
`;

exports[`SelectionControl tests should render correctly with almost all selected items 1`] = `
<styled.div>
  <styled.span>
    3 itens foram selecionados
  </styled.span>
  <Styled(styled.span)
    onClick={[Function]}
  >
    <Styled(Icon)
      path="M0.41,13.41L6,19L7.41,17.58L1.83,12M22.24,5.58L11.66,16.17L7.5,12L6.07,13.41L11.66,19L23.66,7M18,7L16.59,5.58L10.24,11.93L11.66,13.34L18,7Z"
    />
    Selecionar todos os 4 itens
  </Styled(styled.span)>
  <Styled(styled.span)
    onClick={[Function]}
  >
    <Styled(Icon)
      path="M19.36,2.72L20.78,4.14L15.06,9.85C16.13,11.39 16.28,13.24 15.38,14.44L9.06,8.12C10.26,7.22 12.11,7.37 13.65,8.44L19.36,2.72M5.93,17.57C3.92,15.56 2.69,13.16 2.35,10.92L7.23,8.83L14.67,16.27L12.58,21.15C10.34,20.81 7.94,19.58 5.93,17.57Z"
    />
     Limpar seleção
  </Styled(styled.span)>
  <styled.div />
</styled.div>
`;

exports[`SelectionControl tests should render correctly with some selected items 1`] = `
<styled.div>
  <styled.span>
    3 itens foram selecionados
  </styled.span>
  <Styled(styled.span)
    onClick={[Function]}
  >
    <Styled(Icon)
      path="M0.41,13.41L6,19L7.41,17.58L1.83,12M22.24,5.58L11.66,16.17L7.5,12L6.07,13.41L11.66,19L23.66,7M18,7L16.59,5.58L10.24,11.93L11.66,13.34L18,7Z"
    />
    Selecionar todos os 10 itens
  </Styled(styled.span)>
  <Styled(styled.span)
    onClick={[Function]}
  >
    <Styled(Icon)
      path="M19.36,2.72L20.78,4.14L15.06,9.85C16.13,11.39 16.28,13.24 15.38,14.44L9.06,8.12C10.26,7.22 12.11,7.37 13.65,8.44L19.36,2.72M5.93,17.57C3.92,15.56 2.69,13.16 2.35,10.92L7.23,8.83L14.67,16.27L12.58,21.15C10.34,20.81 7.94,19.58 5.93,17.57Z"
    />
     Limpar seleção
  </Styled(styled.span)>
  <styled.div />
</styled.div>
`;

exports[`SelectionControl tests should render nothing with no props 1`] = `""`;
